.github/workflows/keep-alive.yml

```diff
@@ -0,0 +1,36 @@
+name: Keep workflows alive
+
+on:
+  # Run on the first of each month at 1:23 UTC
+  schedule:
+  - cron:  '23 1 1 * *'
+  workflow_dispatch:
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Create a minimal payload
+      run: |
+        mkdir _pass
+        touch _pass/pass.txt
+
+    - name: Keep workflow alive
+      uses: peaceiris/actions-gh-pages@v3
```

Comment:
I think the security risk of using a `@v3` tag here is probably fine as we aren't deploying thing, but as we _are_ pushing back to the repo I can see some motivation to go with a more security cautious approach. I'll defer to others here.

Reply:
I don't see why this is less risky than other cases: the action gets your secret and can push a new tag to do anything with it.

Reply:
I think what @matthewfeickert is `@v3` pins to any v3, while `@v3.9.1` or `@<commit-ish>` would pin to a given version so if upstream is compromised we are safe.

Reply:
Presumably v3.9.1 is a tag, so only hashes are secure in this sense?

Reply:
> the action gets your secret and can push a new tag to do anything with it.

The [default `GITHUB_TOKEN` secret](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) created by GitHub Actions at runtime has its permissions limited to the repository that contains the workflow. So it can't perform arbitrary actions. Though, it can write (and so rewrite) commits, so given that

> so only hashes are secure in this sense?

yes, probably.